compiler/il/OMRNode.cpp

```diff
@@ -2388,56 +2388,49 @@ OMR::Node::computeIsCollectedReference()
-         case TR::aload:
-         case TR::loadaddr:
-            {
-            TR::Symbol *symbol = curNode->getSymbolReference()->getSymbol();
-            // isCollectedReference() responds false to generic int shadows because their type
-            // is int. However, address type generic int shadows refer to collected slots.
-            if (opValue == TR::aloadi && symbol == TR::comp()->getSymRefTab()->findGenericIntShadowSymbol())
-               return true;
-            else
-               return symbol->isCollectedReference();
-            }
-         case TR::aconst:
-            // aconst null can either be collected or uncollected. The collectedness of aconst
-            // null can be determined by usedef analysis, which is too expensive for a very
-            // basic query. Treating aconst null as a collected reference is always safe. The
-            // only problem is: we might mark an aladd/aiadd on a temp whose value is null as
-            // an internal pointer, thus creating a temp, pinning array and internal pointer
-            // map for it.
-            return self() == curNode && curNode->getAddress() == 0;
-         case TR::getstack:
-            return false;
-         default:
-            TR_ASSERT(false, "Unsupported opcode %s on node " POINTER_PRINTF_FORMAT, op.getName(), curNode);
-            return false;
+         curNode = curNode->getFirstChild();
+         continue;
+         }
+
+      // opcodes associated with a symref, we should
+      // be able to tell its collectedness from its symref.
+      if (op.isLoadVar() || op.isLoadAddr() || op.isLoadReg())
+         {
+         TR::Symbol *symbol = curNode->getSymbolReference()->getSymbol();
+         // isCollectedReference() responds false to generic int shadows because their type
+         // is int. However, address type generic int shadows refer to collected slots.
+         if (opValue == TR::aloadi && symbol == TR::comp()->getSymRefTab()->findGenericIntShadowSymbol())
+            return true;
+         else
+            return symbol->isCollectedReference();
          }
+
+      // Symbols for calls and news does not contain collectedness information.
+      // Current implementation treats all object references collectable, and also
+      // assumes that the return of an acall* is an object reference.
+      if (op.isNew() || op.isCall() || opValue == TR::variableNew || opValue == TR::variableNewArray)
+         return true;
+
+      // aconst null can either be collected or uncollected. The collectedness of aconst
+      // null can be determined by usedef analysis, which is too expensive for a very
+      // basic query. Treating aconst null as a collected reference is always safe. The
+      // only problem is: we might mark an aladd/aiadd on a temp whose value is null as
+      // an internal pointer, thus creating a temp, pinning array and internal pointer
+      // map for it.
+      if (opValue == TR::aconst)
```

Comment:
Added a switch statement for the singled out opCodeValues as suggested to make the code cleaner. `TR::variableNew` and `TR::variableNewArray` doesn't have the isNew property, I leave these 2 with the call and new since they are logically grouped together.